### PR TITLE
feat: lending quote dynamic components

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Add borrow/input dynamic components on quote.

Additionally, fixes the wrong debt update for borrows in `<LoanSummary />`

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None

## Testing

- Updated values on quote look sane

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

https://github.com/shapeshift/web/assets/17035424/81649e45-e37d-4af1-929e-62fef5b5fba2

### CODE REVIEW

File: src/pages/Lending/Pool/Pool.tsx [hunk 1]
```diff
@@ -67,9 +72,27 @@ const flexDirPool: ResponsiveValue<Property.FlexDirection> = { base: 'column', l
 type MatchParams = {
   poolAccountId?: AccountId
 }
+
+// Since dynamic components react on the `value` property, this wrappers ensures the repayment lock
+// component accepts it as a prop, vs. <Skeleton /> being the outermost component if not using a wrapper
+const RepaymentLockComponentWithValue = ({ isLoaded, value }: AmountProps & SkeletonOptions) => (
+  <Skeleton isLoaded={isLoaded}>
+    <RawText fontSize='2xl' fontWeight='medium'>
+      {value ?? '0'} days
```
Comment (gomesalexandre) [inline src/pages/Lending/Pool/Pool.tsx:81]: Already a string

Comment (gomesalexandre) [reply src/pages/Lending/Pool/Pool.tsx]: 0129ca2fa8

File: src/pages/Lending/Pool/components/Repay/Repay.tsx [hunk 2]
```diff
@@ -12,27 +12,38 @@ const RepayEntries = [RepayRoutePaths.Input, RepayRoutePaths.Confirm]
...
+  )
 
   return (
     <MemoryRouter initialEntries={RepayEntries} initialIndex={0}>
       <RepayRoutes
+        repaymentAsset={repaymentAsset}
+        setRepaymentAsset={setRepaymentAsset}
         collateralAssetId={collateralAssetId}
         repaymentPercent={repaymentPercent}
         onRepaymentPercentChange={handleDepositAmountChange}
```
Comment (woodenfurniture) [inline src/pages/Lending/Pool/components/Repay/Repay.tsx:49]: might be able to remove the wrapped callback

```suggestion
  return (
    <MemoryRouter initialEntries={RepayEntries} initialIndex={0}>
      <RepayRoutes
        repaymentAsset={repaymentAsset}
        setRepaymentAsset={setRepaymentAsset}
        collateralAssetId={collateralAssetId}
        repaymentPercent={repaymentPercent}
        onRepaymentPercentChange={setRepaymentPercent}
```

Comment (gomesalexandre) [reply src/pages/Lending/Pool/components/Repay/Repay.tsx]: 🧠  https://github.com/shapeshift/web/pull/5679/commits/a9e114696c9411e315d72ef1456aee41743a8569

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**